pepper-apis/src/main/java/org/broadinstitute/ddp/event/publish/pubsub/EventActionPubSubPublisher.java

```diff
@@ -0,0 +1,120 @@
+package org.broadinstitute.ddp.event.publish.pubsub;
+
+import static java.lang.String.format;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_TASK_TYPE;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutureCallback;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
+import com.typesafe.config.Config;
+import org.broadinstitute.ddp.constants.ConfigFile;
+import org.broadinstitute.ddp.event.publish.EventActionPublisher;
+import org.broadinstitute.ddp.housekeeping.PubSubConnectionManager;
+import org.broadinstitute.ddp.model.activity.types.EventActionType;
+import org.broadinstitute.ddp.util.ConfigManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link EventActionPublisher} providing publishing
+ * an event of a specified type to Google PubSub topic.<br>
+ * The parameters for PubSub publishing are taken from config.
+ * <b>Event publishing config parameters</b>:
+ * <ul>
+ *     <li>`usePubSubEmulator` - (true or false)</li>
+ *     <li>`googleProjectId` - project ID (for a pubsub topic)</li>
+ *     <li>`pubsub.eventActionTopic` - name of PubSub topic where to publish event</li>
+ * </ul>
+ */
+public class EventActionPubSubPublisher implements EventActionPublisher {
```

Comment:
Why do we have to create a special "event action" publisher? Isn't there a general "pubsub task" publisher we can use?

Reply:
The existing PubSubTask service - for opposite purpose (it used to consume tasks from DSM to be executed on DSS side). And the current ticket: to send a task from DSS for DSM (to be consumed by DSM).

Reply:
The way I had been thinking about it is that we just have a "communication channel" between DSS and DSM, and we're passing messages around. The message could be a "result for update_profile" or it could be "please update_custom_workflow". Maybe I'm misunderstanding things here.

Reply:
> "communication channel" between DSS and DSM
@yufengwng I strongly agree it is a good idea to have a single communication channel: may be we could discuss it with DSM team and redesign it such way that 2 topics are used for communication between DSM and DSS, but it require some redesign on DSS and DSM sides. Unfortunately we not discussed common design (some common approach - I mentioned this problem during some of "Retro" meetings): currently we design features separately. But I think that it could be reused the same code for both systems..

Yet another understanding of PubSubTasks exist: it is not tasks for DSS from DSM only but from any system from outside world. In theory a task for DSS could be published to "incoming PubSub topic" by any external application (and this external app can process response posted by DSS to "result topic"). So, PubSubTask - an instrument to command by DSS from outside.
In current task we do similar thing but for DSM (to command by it from outside - and such commands could be assigned not only by DSS but any app having access to the dedicated PubSub topics)...
So, it requires another pair of topics to command DSM and process its results (though in the current events processing task we do not used result topic - to check results from DSM)...
Of course we can reuse the same 2 topics for both apps (for commands for DSS and for commands for DSM) but it needs some additional attribute in it.. May be we could define more general design in some next iteration?
And it is definitely should be used same code for both systems as it does same things. Now it is completely separate code..